### PR TITLE
[ee8] WFLY-6463 Common Annotations 1.3 Support

### PR DESCRIPTION
**Pull request by ctomc**
https://issues.jboss.org/browse/WFLY-6463
https://issues.jboss.org/browse/EAP7-900

### CODE REVIEW

**Comment by kabir** [comment]
@ctomc This needs a rebase now. Also n1hility, I am not that familiar with this - do we need to track this in EAP7?

**Comment by scottmarlow** [comment]
@ctomc I ran the EE 8 signature tests with this change and that seemed okay, I didn't try the same with EE 7 yet, will do that today.

**Comment by scottmarlow** [comment]
@ctomc looks like we need both the EE 7 common annotation jar and EE 8 common annotation jar, depending on whether we are in EE 7 or EE 8 mode.

**Comment by kabir** [comment]
@scottmarlow IOW we need module switching?

**Comment by scottmarlow** [comment]
@kabir yes, we need module switching

**Comment by kabir** [comment]
@ctomc See http://post-office.corp.redhat.com/archives/jboss-as/2018-February/msg00011.html for (discussion of) what is needed

**Comment by scottmarlow** [comment]
@kabir yes, sounds like we need module switching as well.

**Comment by ctomc** [comment]
@kabir updated PR to use version switching based on system property.

**Review by jamezp — APPROVED** [review]

File: ee/pom.xml [hunk 1]
```diff
@@ -83,7 +83,7 @@
         </dependency>
         <dependency>
             <groupId>org.jboss.spec.javax.annotation</groupId>
-            <artifactId>jboss-annotations-api_1.2_spec</artifactId>
+            <artifactId>jboss-annotations-api_1.3_spec</artifactId>
```
Comment (jamezp) [inline ee/pom.xml]: I approved this, but what's the rational for changing these?

Comment (ctomc) [reply ee/pom.xml]: we want 1.3 to be used everywhere given this is something that we are going use going forward.
only at runtime we expose different version.
Given the change set between two, it is harmless for pretty much all scenarios (but specific tests)

**Review by jamezp — CHANGES_REQUESTED** [review]
The commit to remove the CLI script need to be deleted from this PR. We will want this once the WIldFly Core is merged and we will remove the argument from surefire.

For the second commit we either need to remove it or fix the test because it's not passing.

**Comment by ctomc** [comment]
@jamezp fixed and removed

**Review by jamezp — APPROVED** [review]

**Pull request merged**